operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,460 @@
+- Configure Loki to span data replication across multiple zones.
+
+## Motivation
+
+Zone-aware replication is the replication of data across failure domains. Avoiding data loss during a domain outage is the motivation to introduce a zone-aware component deployment and enable Loki's zone-aware data replication capabilities.
+
+### Goals
+
+- The user can enable zone aware replication in the loki operator
+- The LokiStack administrator can enable zone-aware data replication for a managed Loki cluster.
+- The LokiStack administrator can choose at least one topology label to enable the cluster spread across domains
+
+### Non-Goals
+
+- Minimize cross-zone bandwidth costs.
+- Introduce data replication across Loki clusters in different regions.
+
+## Proposal
+
+The following enhancement proposal describes the required API additions and changes in the Loki Operator to add zone-aware data replication support.
+
+### API Extensions
+
+#### LokiStack Changes: Support for configuring zone-aware data replication
+
+The following API changes introduce a new spec to enable configuration for Loki's data replication properties.
+
+_Note:_ The new `replicationSpec` introduces in addition a field `factor` that is a replacement for the old `replicationFactor` field. Moving forward the old field is officially deprecated any will be removed in future CRD versions.
+
+```go
+
+import "github.com/prometheus/prometheus/model/labels"
+
+
+//  LokiStackSpec defines the desired state of LokiStack
+type LokiStackSpec struct {
+...
+    // ReplicationFactor defines the policy for log stream replication. (Deprecated: Please use replication.factor instead. This field will be removed in future versions of this CRD)
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:validation:Minimum:=1
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Replication Factor"
+    ReplicationFactor int32 `json:"replicationFactor,omitempty"`
+ 
+    // Replication defines the configuration for Loki data replication
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    Replication *ReplicationSpec `json:"replication,omitempty"`
+}
+
+type ReplicationSpec struct {
+    // Factor defines the policy for log stream replication.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:validation:Minimum:=1
+    Factor int32 `json:"factor,omitempty"`
+
+    // Zone is the key that defines a topology in the Nodes' labels
+    //
+    // +required
+    // +kubebuilder:validation:Required
+    Zones []ZoneSpec
+}
+
+// ZoneSpec defines the spec to support zone aware component deployments.
+type ZoneSpec struct {
+    // MaxSkew describes the maximum degree to which Pods can be unevenly distributed
+    //
+    // +optional
+    // +kubebuilder:validation:optional
+    // +kubebuilder:default:="1"
```

Comment:
I don't think it makes sense to define both "optional" and "default" as the default will always be set.

Reply:
Changed it to required